### DISCUSSION TITLE
從0️⃣開始利用虛擬機建立 Linux 環境

### DISCUSSION THREAD

**Discussion by pudding0803** [1101-GS4719-C/Python-Community-GS4719-C#1454] [Linux]
## 前言

> 想嘗試操作 Linux 系統，但電腦的作業系統卻是 Windows 或 MacOS 嗎🥹？

👉以下教學適合從來沒嘗試過的人，非常新手導向🥳

🛠️此範例教學使用之工具：
- [VirtualBox](https://www.virtualbox.org/wiki/Downloads)
- [Ubuntu](https://www.ubuntu-tw.org/modules/tinyd0/)

⏰此範例建立之時間：2023/05/20
> 未來若官網或版本有更動，則此教學不一定適用

## 下載
- VirtualBox
依照自己現在的作業系統選擇
![image](https://github.com/NCU-GS4719-Python/Python-Community/assets/45879162/b6879e66-5d7f-45a4-afc7-800695ced2d6)

- Ubuntu
基本上都照預設就好，如果電腦是 32 位元則要記得選 32 位元，這會下載滿久的
![image](https://github.com/NCU-GS4719-Python/Python-Community/assets/45879162/bbb82898-438b-42cf-8a61-ccee962df607)

## 建立環境教學

1. 下載並安裝完成 VirtualBox 之後直接打開，按下「新增」
![image](https://github.com/NCU-GS4719-Python/Python-Community/assets/45879162/1b554f22-6c86-4b30-8f69-3585f23d5eda)

2. 輸入想要的虛擬機名稱，其他基本上照預設就好，唯獨 ISO 映像請選擇剛才在 Ubuntu 網站下載的檔案
![image](https://github.com/NCU-GS4719-Python/Python-Community/assets/45879162/064f0ddb-dcac-405f-b636-da65deebf1f1)

3. 下一步，設定使用者名稱與密碼

4. 下一步，如果 RAM 和 CPU 夠多，可以分多一點給虛擬機

5. 跳到完成之後，虛擬機會自動開啟，接著等它安裝一段時間，會滿久的

6. 設定完成之後，就可以使用 Linux 系統了！（之後是在 VirtualBox 按「啟動」就會打開）

### 擷取鍵盤
如果 VirtualBox 有詢問是否要擷取鍵盤請按「擷取」

### 讓全螢幕更好看
在桌面按右鍵▶️「Display Settings」▶️更改 Resolution 成符合螢幕大小的比例
而在 VirtualBox 上方的「檢視」可以更改視窗大小

### 更改語言
在「Settings」選擇「Resion & Language」可以更改語言

### 與主機共用
在 VirtualBox 上方的「裝置」可以設定「共用資料夾」、「共用剪貼簿」、「拖放」等
這會使得「在主機和虛擬機之間移動文字或檔案等」變得更加方便

## 問題與討論
🌟歡迎沒試過的可以嘗試看看，有問題或心得也可以在下方留言👇，我會盡我所能🥹

❌如果有錯誤也歡迎指出，謝謝大家😇！

**Comment by TW-Maverick** [comment]
記得好像還有另外兩種方法，各自有優缺點，我是使用第二種方法，這樣就不需自行劃出硬碟空間，而且允許使用者在Windows環境下即可同時擁有使用Linux命令工具和利用Linux開發環境的能力。我覺得這是子系統最大的好處，但見仁見智！

1. Windows & Ubuntu 22.04單碟雙系統

2. Windows 子系統 Linux 版


🔗[Windows 子系統說明](https://learn.microsoft.com/zh-tw/windows/wsl/about)
🔗[Windows 11安裝Ubuntu 22.04單碟雙系統](https://ivonblog.com/posts/windows11-ubuntu2204-dual-boot/)

**Reply by pudding0803** [reply]
我也有裝這個但不太熟🤣，但我想說虛擬機或許對新手比較友善，至少弄壞了不會影響到電腦(?)

謝謝分享🥳！
